File: internal/service/balanceService.go
package service

import "wasd0/is-rest/api/dto"

type BalanceService interface {
	Create(request dto.BalanceCreateRequest) (dto.BalanceCreateResponse, *dto.ServErr)
	GetByCustomer(request dto.BalanceGetRequest) (dto.BalanceGetResponse, *dto.ServErr)
	GetById(id int64) (dto.BalanceGetResponse, *dto.ServErr)
}

File: src/main/kotlin/com/wasd/isrest/service/BalanceService.kt
package com.wasd.isrest.service

import com.wasd.isrest.domain.Balance
import com.wasd.isrest.model.request.BalanceCreateRequest
import com.wasd.isrest.model.request.BalanceGetRequest
import com.wasd.isrest.model.response.BalanceCreateResponse
import com.wasd.isrest.model.response.BalanceGetResponse

interface BalanceService {

    fun create(request: BalanceCreateRequest): BalanceCreateResponse

    fun getDataByCustomerAndCurrency(request: BalanceGetRequest): BalanceGetResponse

    fun getByCustomerIdAndCurrencyCode(customerId: Long, currencyCode: String): Balance

    fun getById(id: Long): BalanceGetResponse
}